xir/parser.py

```diff
@@ -174,70 +173,28 @@ def expr(self, args):
     # definitions and statements
     ##############################
 
-    def gate_def(self, args):
+    @v_args(inline=True)
+    def gate_def(self, name, params_list, wires, *stmts):
         """Gate definition. Starts with keyword 'gate'. Adds gate to program."""
-        name = args.pop(0)
-        wires = ()
-        params = []
-        stmts = []
 
-        max_wire = 0
         has_declared_wires = False
-        for i, arg in enumerate(args):
-            if is_param(arg):
-                params = arg[1]
-            elif is_wire(arg):
-                has_declared_wires = True
-                wires = arg[1]
-            elif isinstance(arg, Statement):
-                if has_declared_wires:
-                    stmts = args[i:]
-                    break
-                stmts.append(arg)
-
-                int_wires = [w for w in arg.wires if isinstance(w, int)]
-                if int_wires and max(int_wires) > max_wire:
-                    max_wire = max(int_wires)
+        if wires is not None and len(wires) > 0:
```

Comment:
I think this suffices:
```suggestion
        if wires:
```